### PR TITLE
Use upstream pkl-gha for GitHub Action Workflows

### PR DESCRIPTION
**Pull request by StefMa**
## Description

Hello Realm .NET Team 👋

It’s great to see that you’re considering using Pkl, especially pkl-gha, to configure your CI 🙂. 
I’ve incorporated all of your additions into the upstream project, so your customizations are no longer necessary. 
This means there’s no longer any maintenance overhead required on your end.

I’ve split the PR into three commits:

1.	Adopts the upstream project.
2.	Removes your custom Pkl project.
3.	Translates the Pkl workflows into YAML workflows.

Each of these commits can be reviewed individually.

Please note that I haven’t changed any of your logic.
I’ve only replaced your implementation with the upstream project.

I hope you find this useful! 
Let me know if anything is missing or if there are any improvements I can make to pkl-gha. 
Looking forward to your review and feedback 🤝.

cc @nirinchev because you added pkl at the first place here https://github.com/realm/realm-dotnet/pull/3603 🙂

### CODE REVIEW

File: .github/build-workflows.ps1 [hunk 1]
```diff
@@ -1,3 +1 @@
-Get-ChildItem -Path './pkl-workflows' -Filter *.pkl -File -Name | ForEach-Object {
-    &pkl eval ./pkl-workflows/$_ -o "./workflows/$($_.Replace('pkl', 'yml'))"
-}
\ No newline at end of file
+pkl eval pkl-workflows/*.pkl -o workflows/%{moduleName}.yml
```
Comment (StefMa) [inline .github/build-workflows.ps1:1]: I don't know if this is correct.
Please check this if this still works.
I'm not a Windows user...

**Comment by nirinchev** [comment]
Hey @StefMa thanks so much for this - great stuff. I was going to eventually try and submit a PR upstream for those changes, but priority reshuffles within the company meant I don't have that much time working on this project anymore. I'll try and review this over the weekend and get it in.

**Comment by StefMa** [comment]
@cla-bot check

**Comment by StefMa** [comment]
@nirinchev what weekned did you mean? 🤔 

Just kidding 🙂 
Take your time. This should act only as a friendly reminder...

**Comment by nirinchev** [comment]
Hey @StefMa, was just planning to get to it after I put some final touches on the Half Life 3 release - should be out any day now 😝

It's a fair callout - been distracted lately with other things, but will try and make time for it [Very Soon™](https://wowwiki-archive.fandom.com/wiki/Soon#Very_Soon™)!

**Pull request merged**